### PR TITLE
Filter by relationship params and get it

### PR DESCRIPTION
**Pull request by MardanovTimur**
```
some_isntance.city.filter_relationsips(some_city_instance, {relationship_filter_kwargs}) -> returns relationships
```

### CODE REVIEW

**Comment by MardanovTimur** [comment]
@robinedwards  it'is works too.

**Comment by MardanovTimur** [comment]
image

**Comment by robinedwards** [comment]
Hi thanks for this! There's actually already a method for this called 'match' which I dont particularly like the name. As we are getting close to a 4.0.0 release, We could rename it to filter_relationships and add a deprecation note to the old one?

**Comment by MardanovTimur** [comment]
No, match returns the nodes whose filter by relationships. filter_relationships returns relationships, filtering by rels too

File: neomodel/match.py [hunk 1]
```diff
@@ -125,6 +126,9 @@ def install_traversals(cls, node_set):
         traversal = Traversal(source=node_set, name=key, definition=rel.definition)
         setattr(node_set, key, traversal)
 
+re._alphanum_str = frozenset(
+    "_абвгдеёжзийклмнопрстуфхцчшщъыьэюяАБВГДЕЁЖЗИЙКЛМНОПРСТУФХЦЧШЩЪЫЬЭЮЯ\
+            abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ01234567890")
```
Comment (robinedwards) [inline neomodel/match.py]: I think theres an actual regex property for this

File: neomodel/match.py [hunk 2]
```diff
@@ -264,22 +267,27 @@ def build_traversal(self, traversal):
...
-
-        # build source
+        #build source
         lhs_ident = self.build_source(traversal.source)
-        rhs_ident = traversal.name + rhs_label
-        self._ast['return'] = traversal.name
+        rhs_ident = "{alias}:{def_label}".format(alias=traversal.name,
+                def_label=traversal.definition.get('node_class').__label__)
+
+
```
Comment (robinedwards) [inline neomodel/match.py]: extra white spaxe

File: neomodel/match.py [hunk 3]
```diff
@@ -264,22 +267,27 @@ def build_traversal(self, traversal):
         """
         traverse a relationship from a node to a set of nodes
         """
-        # build source
-        rhs_label = ':' + traversal.target_class.__label__
-
-        # build source
+        #build source
```
Comment (robinedwards) [inline neomodel/match.py]: space after #the

File: neomodel/match.py (hunk 1 above)
Comment (MardanovTimur) [reply neomodel/match.py]: It does not work with the Cyrillic alphabet. Now it works. Before Cyrillic symbols re.escape insert 2 backslashes and neomodel build incorrect query.

File: neomodel/match.py (hunk 2 above)
Comment (MardanovTimur) [reply neomodel/match.py]: Ok, i will fix it now

File: neomodel/match.py (hunk 3 above)
Comment (MardanovTimur) [reply neomodel/match.py]: ok

**Comment by MardanovTimur** [comment]
@robinedwards ready

File: neomodel/match.py (hunk 1 above)
Comment (robinedwards) [reply neomodel/match.py]: https://docs.python.org/2/library/re.html#re.UNICODE we just need to set this flag where the regex is being used. Can you do that and put it in a separate MR from this one please. Thanks!

**Comment by robinedwards** [comment]
I think `match(some_property='bar').all_relationships()` maybe what you are after?

**Comment by whatSocks** [comment]
will check on in https://github.com/neo4j-contrib/neomodel/issues/594

**Pull request closed without merging**